app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -91,25 +86,44 @@ def callback_options
     { 'object_global_id' => object.to_global_id, 'caller_worker_hierarchy' => caller_worker_hierarchy }
   end
 
-  def delete_objects_of_association(main_object, association_name, worker)
-    return unless (associated_objects = main_object.public_send(association_name))
-    if associated_objects.respond_to?(:each)
-      associated_objects.each { |associated_object| delete_association_perform_later(worker, associated_object) }
-    else
-      delete_association_perform_later(worker, associated_objects)
+  class ReflectionDestroyer
+    def initialize(main_object, reflection, caller_worker_hierarchy)
+      @main_object = main_object
+      @reflection = reflection
+      @caller_worker_hierarchy = caller_worker_hierarchy
     end
-  end
 
-  def associations_to_destroy_for(record)
-    record.class.reflect_on_all_associations.select { |reflection| reflection.options[:dependent] == :destroy }
-  end
+    def destroy_later
+      return unless reflection.options[:dependent] == :destroy
+      reflection.macro == :has_many ? destroy_has_many_association : destroy_has_one_association
+    end
 
-  def delete_association_perform_later(worker, object)
-    worker.perform_later(object, caller_worker_hierarchy)
-  end
+    attr_reader :main_object, :reflection, :caller_worker_hierarchy
 
-  def association_delete_worker(reflection)
-    delete_hierarchy_workers = Hash.new(DeleteObjectHierarchyWorker).merge('Account' => DeleteAccountHierarchyWorker)
-    delete_hierarchy_workers[reflection.options[:class_name]]
+    private
+
+    def destroy_has_many_association
+      main_object.public_send(reflection.name).ids.each do |associated_object_id|
```

Comment:
```suggestion
      main_object.public_send("#{reflection.name}_ids").each do |associated_object_id|
```

Reply:
@hallelujah that doesn't work
![image](https://user-images.githubusercontent.com/11318903/60906875-7a1ce500-a278-11e9-8448-3b0a1b93cd52.png)

Reply:
ok, this works:
```ruby
main_object.public_send("#{reflection.name.to_s.singularize}_ids").each do |associated_object_id|
```